### DISCUSSION TITLE
4.0.0-beta.1 ARM 64(Native) Feedback

### DISCUSSION THREAD

**Discussion by linalan** [transmission/transmission#3903] [Support]
First, great to finally see a new version, especially with support for Apple Silicon. Many thanks to all involved! 

I know it's beta! That said, I'm experiencing some issues. Actually same issues I've been experiencing w/the nightly builds I've tried over the last couple months. 

- Downloads  never seem to finish. It will say 100%, but never change color nor will the file be completely written (i.e., remains xxxxx.mkv.part) Occasionally, if I shut Transmission down and restart it, the file will go green and be marked/written as complete.
- Constant file corruption. Never in all my years w/Transmission on Intel Mac have I seen it tell me to "verify local data".
- Random crashing

Again, I saw these same issues w/the nightly builds. Downloading the same torrent on the Intel Mac, quick and painless. 

**Machine Specs**
- Macbook Pro 14
- M1 Pro Chip
- macOS 12.6
- 16GB RAM

[crash-report.txt](https://github.com/transmission/transmission/files/9738616/crash-report.txt)

**Comment by TheFalcon81** [comment]
I am also having issues with Transmission since moving to a M1 Pro Macbook Pro. It seems to struggle to connect with peers. I thought it may have been network settings however, after much time checking and rechecking I have determined it is the 4.00 client. I tried the version of Transmission running on my Synology NAS and I was able to successfully download a torrent that my Macbook Pro couldn't. I also installed qBittorrent for the first time and tested the same torrent on the same M1 Pro mac and it worked. Any thoughts or is this a known issue? I appreciate that this is still in Beta however its currently unusable in my circumstances.

**Reply by TheFalcon81** [reply]
mine seems to have fixed itself after closing the transmission overnight. Not sure why that would make a difference. I also turned off "Automatically Map Port" since I am port forwarding to an assigned port through my VPN provider. Been solid so far.

**Reply by TheFalcon81** [reply]
Still broken, I have reverted back to 3.0 Stable for Mac as the Nightly builds work one day, and then suddenly they struggle to connect and nothing downloads. It's frustrating as I can't figure out why it intermittently fails. I have seen other user reports of the same issue

**Reply by TheFalcon81** [reply]
see https://github.com/transmission/transmission/issues/3285#issue-1269635092

**Comment by ckerr** [comment]
Crash report:

```
6   libc++abi.dylib               	       0x187115ea4 std::__terminate(void (*)()) + 20
7   libc++abi.dylib               	       0x187115e40 std::terminate() + 64
8   Transmission                  	       0x1044ce0b8 0x104428000 + 680120
9   Transmission                  	       0x1044d0fbc 0x104428000 + 692156
10  Transmission                  	       0x1044ce71c 0x104428000 + 681756
11  Transmission                  	       0x104430c1c 0x104428000 + 35868
12  CoreFoundation                	       0x18721d468 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 28
```

@mikedld it looks like debugging info was stripped out of 4.0.0-beta.1...? Can that be restored for other betas so that these crash reports are more useful?

@linalan could you try a nightly build from https://build.transmissionbt.com/job/trunk-mac/ to see if the crash persists / to see if we can get a more productive crash report?

**Reply by linalan** [reply]
@ckerr I did switch back to the  nightly builds a week or so ago. I haven't had a crash since 🤷‍♂️ Go figure! However, I'm still seeing some other issues, i.e., torrents never transitioning to completed even though showing 100% and exceeding seed ratio, lots of file corruption

**Comment by Coeur** [comment]
@ckerr I support having a beta 2 now, without delay, with the present code. Essential changes have been made, and the most important one being that the beta 1 release for macOS was almost unusable as torrents would be paused without possibility to restart them (#3882 and #3907). Thank you

**Reply by ckerr** [reply]
@Coeur soon...

**Comment by Scorpys49** [comment]
hi,
works on intel iMac 2019 , but transmission can't open any port , all is close or timeout with ou without UPNP , test with qbittorent all work with it 

All my open is open in my box
